src/index.ts

```diff
@@ -32,11 +32,11 @@ const _hasOwn = Object.prototype.hasOwnProperty;
  * @template T
  */
 export function map<T>(opt_initial: T | undefined) {
-  const obj = Object.create(null);
+  const object = Object.create(null);
   if (opt_initial) {
-    Object.assign(obj, opt_initial);
+    Object.assign(object, opt_initial);
   }
-  return obj;
+  return { ...opt_initial };
```

Comment:
I believe this changes the fact that _every_ object being returned from this function would have a `null` prototype, but w/ this solution, it seems like that would no longer be the case&hellip;

That would seem like a breaking change to me, and am unsure if it is desirable, but will leave this here as-is so we can determine what we think we should do (null prototypes are more lightweight but less natural).

```suggestion

  // FIXME(@DerekNonGeneric): Should we be creating objects w/ null protos?
  return { ...opt_initial };
```